File: ledcommsbrightness.cpp
/***

Embed microcontroller serial port listener file
that shall listen to the host PC for bytes sent over.
Depending on the message, the LED on the microcontroller
will either light up or dim down. The microcontroller shall
response with what action was performed and send the message
back to the host PC.

***/ 

#include "mbed.h"
 
Serial pc(USBTX, USBRX); // tx, rx
PwmOut led(LED1);
 
float brightness = 0.0;
 
int main() {
        /* Listen for characters sent
        from host PC and checks whether to
        increase the LED's brightness or decrease it.
        Microcontroller shall return a response back to
        the host PC to inform the action executed.
        */

    pc.printf("Press '+' to turn LED1 brightness up, '-' to turn it down\n");
    while(1) {
        char c = pc.getc();
        if((c == '+') && (brightness < 0.5)) {
            brightness += 0.01;
            led = brightness;
            pc.putc('+');
        }
        if((c == '-') && (brightness > 0.0)) {
            brightness -= 0.01;
            led = brightness;
            pc.putc('-');
        } 
 
    }
}

File: ledcommsbrightness.py
'''
Communicates with Mbed LPC1768
microcontroller via usb serialport.
Indefinitely waits for user input to
send through serial port and reads from
the port the response from the microcontroller.
Based on the user's input sent to the
microcontroller, LED 1 will either slightly
light up or slightly dim down.
'''
import serial
# Establish connection to the Mbed unit.
ser = serial.Serial('/dev/ttyACM0',
        9600, timeout=2)
def ledcomms():
    # Print instructions from Microcontroller
    print ser.readline()
    while True:
        serial_input = raw_input("Enter '+' or '-':\n")
        ser.write(serial_input + "\n")
        print "Sent string: %s" % serial_input
        print "Response: %s" % ser.readline()

if __name__ == "__main__":
    ledcomms()
